newIDE/app/src/EventsSheet/index.js

```diff
@@ -178,6 +181,31 @@ export default class EventsSheet extends React.Component<Props, State> {
     });
   }
 
+  componentWillMount() {
+    const allExtensions = gd
```

Comment:
So moving this logic here is good, but I'm concerned about this size of `EventsSheet` which is becoming a really large file. Would it be possible to move this (from `const allExtensions...` to the `flatten()` in a file called `EnumerateEventsMetadata.js`? You can have inside this function that is called enumerateEventsMetadata: `export const enumerateEventsMetadata = () => ...`.

Then here in this file in the componentWillMount, you only do:
`this.allEventsMetadata = enumerateEventsMetadata()`

Note that componentWillMount will be deprecated in future versions of React (see in https://reactjs.org/docs/react-component.html). Instead, we can use `state` and componentDidMount:

```
componentDidMount() {
   this.setState({ allEventsMetadata: enumerateEventsMetadata() });
}
```

you have to change the typing of State and finally change the usage of `this.allEventsMetadata` to `this.state.allEventsMetadata` :)

Reply:
Would youlike that file to be in the utility functions folder? :)

Reply:
I think this file can live in the same directory as EventSheet - as it's only used by this component (and the toolbar). This will avoid cluttering Utils folder with things that are not reused. :)

Reply:
All done now :)